From 7c3a822d1a90a5f781f8a3cd960d6f16c069800a Mon Sep 17 00:00:00 2001
From: Evgeny Eltsin <eaeltsin@google.com>
Date: Wed, 4 Aug 2010 21:08:24 +0400
Subject: [PATCH 39/40] [gcc] Prevent movsi from having 2 memory operands in x86_64 PIC.

The cause is that rip-based symbol references are treated as constants with "i" constraint, "e" seems to do the right job.

Review URL: http://codereview.chromium.org/3013058
---
 gcc/gcc/config/i386/i386.md |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gcc/gcc/config/i386/i386.md b/gcc/gcc/config/i386/i386.md
index 7155cbc..0f1407b 100644
--- a/gcc/gcc/config/i386/i386.md
+++ b/gcc/gcc/config/i386/i386.md
@@ -1535,7 +1535,7 @@
   [(set (match_operand:SI 0 "nonimmediate_operand"
 			"=r,m ,*y,*y,?rm,?*y,*x,*x,?r ,m ,?*Yi,*x")
 	(match_operand:SI 1 "general_operand"
-			"g ,ri,C ,*y,*y ,rm ,C ,*x,*Yi,*x,r   ,m "))]
+			"g ,re,C ,*y,*y ,rm ,C ,*x,*Yi,*x,r   ,m "))]
   "!(MEM_P (operands[0]) && MEM_P (operands[1]))"
 {
   switch (get_attr_type (insn))
-- 
1.7.1

